pypeit/utils.py

```diff
@@ -1238,6 +1238,83 @@ def robust_optimize(ydata, fitfunc, arg_dict, maxiter=10, inmask=None, sigma=Non
                     maxdev=None,maxrej=None, groupdim=None, groupsize=None, groupbadpix=False, grow=0, sticky=True,
                     use_mad=True, **kwargs_optimizer):
 
+    """
```

Comment:
Thanks for adding this doc string, but the format is wrong.

Reply:
Oh, I thought I was copying our preferred format, as I was using the doc for robust_polyfit_djs, to which this code is very similar. We need a piece of code that I can quickly reference that has the doc strings in the correct format. I had to add something non-standard here, since the code takes a function as an argument, and I need to somewhere specify that function's calling sequence. Perhaps that is what you mean by wrong format.

Reply:
Here are two reference docs:
 - https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html
 - https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google

An example in our code base is masterframe.py.

Ultimately what matters is how it displays after Sphinx parses the docstrings and constructs the html.  PypeIt isn't doing well on this front because most of the modules fail the build, but here's example output from masterframe:

![sphinx_output](https://user-images.githubusercontent.com/15785898/58029557-b41c0700-7ad1-11e9-8aae-02ec7424106d.png)

I couldn't find a syntax for documenting callable objects in the argument list.  I would convert that entry to:

```
    fitfunc (callable):
        The callable object used to perform the fitting.  The calling
        sequence must be::

            result, ymodel = fitfunc(ydata, inmask, arg_dict, **kwargs_optimizer)

        See the descriptions of `ydata`, `inmask`, `arg_dict`, and
        `kwargs_optimizer`.

        The two returned objects are:
            - `result`: Object returned by the specific scipy.optimize
              method used to perform the fit.
            - `ymodel`: A `numpy.ndarray` with the model fit to `ydata`
              and with the same shape.
```

I'm not sure that will display properly, but it's my best guess at the moment.

Reply:
Okay done.

Reply:
You fixed that one entry but not the full docstring.  I've done it myself so you have an example.  The formatting and syntax matter here because otherwise Sphinx won't be able to parse the docstring.